metrics/lib/features/dashboard/presentation/widgets/loading_builder.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:flutter/cupertino.dart';
+
+/// Displays the [loadingPlaceholder] widget if the data [isLoading].
+class LoadingBuilder extends StatelessWidget {
+  final bool isLoading;
+  final WidgetBuilder builder;
+  final Widget loadingPlaceholder;
+
+  /// Creates the [LoadingBuilder].
+  ///
+  /// [isLoading] defines if the data is loading or not.
+  /// [loadingPlaceholder] the widget that will be shown when the data [isLoading].
+  /// [builder] is the [WidgetBuilder] which will be called after data is loaded.
+  const LoadingBuilder({
+    Key key,
+    this.isLoading,
+    this.builder,
+    this.loadingPlaceholder,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    if (isLoading) return loadingPlaceholder;
+
+    return builder(context);
+  }
```

Comment:
What do you think about adding the default `loadingPlaceholder` and assert if the `builder` function is `null` to be sure that this build method won't return null and cause an error?

Reply:
I would also suggest adding asserts/@required as well as widget test. I think that we miss some widget tests now - I can create ticket to improve it.